### ISSUE TITLE
Incorrect report `Argument of type "Never" cannot be assigned to parameter`

### ISSUE THREAD

**Issue by letrec** [microsoft/pyright#707]
**Describe the bug**
Pyright reports incorrectly `Argument of type "Never" cannot be assigned to parameter "i" in function "deallocate" of type "int"`

`Type "Never" cannot be assigned to type "int"Pyright (reportGeneralTypeIssues)`
 for the line 19 of the repro program provided below.

**To Reproduce**
Paste the code into VS Code with Pyright enabled.

**Expected behavior**
No error should be reported in this case.

**Screenshots or Code**
```
def allocate() -> int:
    return 0

def deallocate(i: int) -> None:
    pass

def use(i: int) -> int:
    return i

def test() -> int:
    i = allocate()
    try:
        i2 = allocate()
        try:
            return use(i2)
        finally:
            deallocate(i2)
    finally:
        deallocate(i) # line of the false positive report
```

**VS Code extension or command-line**
VS Code extension 1.1.40.

This problem doesn't occur without nested finally blocks.

**Comment by erictraut** [comment]
Thanks for the bug report. 

Finally clauses present a difficult challenge for a static type checker because they are executed "out of order" with respect to the rest of the code flow. Finally blocks need to be executed when an exception is raised, when a return is encountered, or in the fall-through case for the try block. 

This becomes even more challenging for nested finally clauses. Pyright doesn't handle all such cases, but I've augmented it to properly handle the simple cases correctly, including your example above. 

This fix will be in the next published version of Pyright.

**Comment by erictraut** [comment]
This is addressed in Pyright 1.1.41, which I just published.
